### ISSUE TITLE
workshop: animation formats

### ISSUE THREAD

**Issue by helenvarley** [upstage-org/upstage#37]
as a logged-in player, i expect to be able to upload visual animations in multiple formats, as avatars, props or backdrops; and to be able to manipulate these animations on stage, i,e, play, pause, stop. (similar to what we can currently do in UpStage v3)

**Comment by helenvarley** [comment]
this needs testing - what formats work? can we upload animated gifs and if so can they be played and stopped? what other animated formats are possible? 
needs to be documented in the manual and also with a tip at the media upload field, so that people are not frustrated by uploading formats that don't work.

**Comment by vickis** [comment]
If a looped gif is uploaded it should continue looping 
currently they only play through once and stop .... 
EXCEPT when they are viewed as guest 
I made an animated gif avatar of a Go Board to try and record each state - they play through only once on each stage as player 
BUT when viewed as guest they play through once and stop - and then if another if put on stage both play through once and stop and again with a third
![image](https://user-images.githubusercontent.com/4249450/139525267-512cf457-c63f-40e6-b9e5-b87be9a39574.png)

**Comment by vickis** [comment]
this is when the 4th is added
![image](https://user-images.githubusercontent.com/4249450/139525303-4c7909b4-6dee-452f-900d-e7f1834e7f76.png)
and player view 
![image](https://user-images.githubusercontent.com/4249450/139525320-319f17c3-b157-4086-95b6-26202ffcabf3.png)

**Comment by helenvarley** [comment]
actually it should be possible for gif animations to either play once, or to loop continuously - i think with gifs this is determined by how the gif is saved, not by UpStage ... i think???
& certainly it should be seen the same by players and adience, looping or not.
(should this be a separate issue? this issue is about what animation formats in general can be used; maybe better to have a separate issue for details of specific format?)

**Comment by vickis** [comment]
Yes it is a looping animation
Generally unless it fills the frame in terms of the image content then it’s
isn’t the best file format anyway as you can’t achieve transparent
backgrounds (so png suits best for the changing positions of avatars etc)
useful in other settings backdrop or prop potentially though

**Comment by nguyenhongphat0** [comment]
Hi @vickis, the loop attribute of a `gif` image is defined in the file itself, not the website that displaying it. For example, you can create a GIF file that is looped forever, or only once (or 2 times, 3 times then stop). Take a look: 

| Loop once | Loop forever |
|-|-|
| ![loop-once](https://user-images.githubusercontent.com/29925961/139575283-193432cf-a918-4e2e-996e-b79e690d4fdd.gif) | ![loop-forever](https://user-images.githubusercontent.com/29925961/139575281-7f45240b-9948-4f09-a8bc-d9a95fea283e.gif) | 

and it look like your gif file has no loop attribute, that's why it stop at the final frame. To enable the loop attribute you just have to open any gif editor (such as this one I used: https://ezgif.com) and make it loop

![gif_loop](https://user-images.githubusercontent.com/29925961/139575437-d5f76207-314d-4b9b-9b23-1d645a5e474e.png)

**Comment by nguyenhongphat0** [comment]
I would recommend create multiframe media in UpStage by composing multiple static images, rather than uploading a GIF image yourself. The reasons are:
- You can dynamically change the animation speed, and play/stop at any frame that you like
- When a player move/resize the avatar with the gif, the animation start over again, it was because the image was `re-render`, which make you think that it is looping for players, while audiences are not

However if these factors don't effect your purpose, GIFs are great and simple to be used. Especially for stage cover images :+1:

**Comment by helenvarley** [comment]
thanks phat - this is useful information to add to the user manual. 

@vicki - it is possible to have transparency in a gif, for example schaumbad's "stay home" avatar is a gif & that has transparency.

**Comment by vickis** [comment]
Awesome thanks Phat :)

**Comment by vickis** [comment]
Ok great :)
